src/getConfig.js

```diff
@@ -0,0 +1,137 @@
+/* eslint no-console: 0 */
+/* eslint no-prototype-builtins: 0 */
+const chalk = require('chalk')
+const format = require('stringify-object')
+const intersection = require('lodash/intersection')
+const defaultsDeep = require('lodash/defaultsDeep')
+const isObject = require('lodash/isObject')
+const validate = require('jest-validate').validate
+const unknownOptionWarning = require('jest-validate/build/warnings').unknownOptionWarning
+const logValidationWarning = require('jest-validate/build/utils').logValidationWarning
+const isGlob = require('is-glob')
+
+/**
+ * Default config object
+ *
+ * @type {{concurrent: boolean, chunkSize: number, gitDir: string, globOptions: {matchBase: boolean, dot: boolean}, linters: {}, subTaskConcurrency: number, renderer: string, verbose: boolean}}
+ */
+const defaultConfig = {
+  concurrent: true,
+  chunkSize: Number.MAX_SAFE_INTEGER,
+  gitDir: '.',
+  globOptions: {
+    matchBase: true,
+    dot: true
+  },
+  linters: {},
+  subTaskConcurrency: 1,
+  renderer: 'update',
+  verbose: false
+}
+
+/**
+ * Check if the config is "simple" i.e. doesn't contains any of full config keys
+ *
+ * @param config
+ * @returns {boolean}
+ */
+function isSimple(config) {
+  return (
+    isObject(config) &&
+    !config.hasOwnProperty('linters') &&
+    intersection(Object.keys(defaultConfig), Object.keys(config)).length === 0
+  )
+}
+
+/**
+ * Custom jest-validate reporter for unknown options
+ * @param config
+ * @param example
+ * @param option
+ * @param options
+ * @returns {void}
+ */
+function unknownValidationReporter(config, example, option, options) {
+  /**
+   * If the unkonwn property is a glob this is probably
+   * a typical mistake of mixing simple and advanced configs
+   */
+  if (isGlob(option)) {
+    const message = `  Unknown option ${chalk.bold(`"${option}"`)} with value ${chalk.bold(
+      format(config[option], { inlineCharacterLimit: Number.POSITIVE_INFINITY })
+    )} was found in the config root.
+  
+  You are probably trying to mix simple and advanced config formats. Adding 
+  
+  ${chalk.bold(`"linters": {
+    "${option}": ${JSON.stringify(config[option], null, '\t    ')}
+  }`)}
+   
+  will fix it and remove this message.`
+
+    const comment = options.comment
+    const name = (options.title && options.title.warning) || 'WARNING'
+    logValidationWarning(name, message, comment)
+  }
+  // If it is not glob pattern, when use default jest-validate reporter
+  return unknownOptionWarning
```

Comment:
`unknownOptionWarning` is a function. So should we be invoking it here instead of returning it?

Reply:
You're right! I thought I've updated that already :-/

Reply:
Mentioned this before but a `return` after `logValidationWarning` is also required.

Reply:
See https://github.com/okonet/lint-staged/pull/141/commits/0d210f4928c5d5050bf6a4f28b44676b4b895985